neurom/analysis/morphtree.py

```diff
@@ -300,6 +300,16 @@ def trunk_azimuth(tree, soma):
     return np.arctan2(vector[COLS.Z], vector[COLS.X])
 
 
+def partition(tree):
+    '''Ratio of maximum number of sections
+       over minimum number of sections at
+       each bifurcation point.
+    '''
+    return [np.max([n_sections(i.children[0]), n_sections(i.children[1])]) /
+            np.min([n_sections(i.children[0]), n_sections(i.children[1])])
```

Comment:
You can define a helper function inside the function, to call n_sections only once per child:

    def partition_at_point(bif_point):
        n = n_sections(bif_point.children[0])
        m = n_sections(bif_point.children[1])
        return max(n, m) / min(n,m)

Also, no idea why you are using `np.max`. All it seems to do is force you to build lists for no reason.